ln-gateway/src/lib.rs

```diff
@@ -101,9 +101,10 @@ where
 {
     async fn handle<F: Fn(T) -> FF, FF: Future<Output = Result<T::Response>>>(self, handler: F) {
         let result = handler(self.request).await;
-        self.sender
-            .send(result)
-            .expect("couldn't send over channel");
+        if self.sender.send(result).is_err() {
+            // TODO: figure out how to log the result
```

Comment:
we once talked with @okjodom about unifying some ln-gateway and clientd functionality, this is another one because clientd has situations like this too (lets talk about this again sometime)